ddtrace/internal/datadog/profiling/dd_wrapper/src/exporter.cpp

```diff
@@ -370,30 +383,23 @@ Profile::start_sample(unsigned int nframes)
 void
 Profile::push_frame_impl(std::string_view name, std::string_view filename, uint64_t address, int64_t line)
 {
-    if (cur_frame >= lines.size() || cur_frame >= locations.size())
+    if (cur_frame >= locations.size())
```

Comment:
How can this happen?

Reply:
This was a rather defensive pattern used as a check because `locations` was a std::array and keeping an array-with-cursor around is somewhat brittle.  This made more sense when we had both lines and locations.

Given the flattening we saw in libdatadog5, I think it makes more sense to refactor the locations buffer as a reserved vector.  This eliminated `cur_frame` as well as `nframes` and simplifies the caller API.  I believe the overall number of destructions for any fixed period of time is equal in the limit either way, so I slightly prefer the simplified code this brought us.